File: backend/app/schemas/sync.py
from pydantic import BaseModel
from typing import Literal, List, Dict, Any, Optional

class SyncOperationItem(BaseModel):
    id: str
    type: str
    patientId: str
    timestamp: str
    payload: Dict[str, Any]

class SyncBatchRequest(BaseModel):
    deviceId: str
    operations: List[SyncOperationItem]

class SyncItemResult(BaseModel):
    operationId: str
    status: Literal['SYNCED', 'FAILED']
    error: Optional[str] = None

class SyncBatchResponse(BaseModel):
    results: List[SyncItemResult]

File: frontend/src/services/api/sync.ts

import { api } from "./client";

export interface SyncOperationItem {
  id: string;
  type: string;
  patientId: string;
  timestamp: string;
  payload: unknown;
}

export interface SyncBatchRequest {
  deviceId: string;
  operations: SyncOperationItem[];
}

export interface SyncItemResult {
  operationId: string;
  status: "SYNCED" | "FAILED";
  error?: string;
}

export interface SyncBatchResponse {
  results: SyncItemResult[];
}

export const syncApi = {
  syncBatch: (request: SyncBatchRequest) =>
    api.post<SyncBatchResponse>("/sync", request),
};
